libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -119,11 +119,13 @@ internal class MapboxTripSession(
             }
             is SetRoutes.RefreshRoutes -> {
                 if (routes.isNotEmpty()) {
-                    isUpdatingRoute = true
                     val primaryRoute = routes.first()
                     lateinit var refreshRouteResult: Expected<String, List<RouteAlternative>>
                     var lastSavedResultValue: Expected<String, List<RouteAlternative>>? = null
-                    for (route in routes) {
+                    // primary route must be refreshed at the very end,
+                    // because after it is refreshed,
+                    // statuses that correspond to the refreshed route will start coming
```

Comment:
Do you think we need a unit test for that case?

Reply:
We already have one: https://github.com/mapbox/mapbox-navigation-android/pull/7399/files#diff-c32f134cc42c85039ed2f86ecb6079636c20232b5d599318c1fe654c06be2034R1541

Reply:
I mean instead of the comment we can add a unit test for the situation that is described in the comment. Scenario: primary route is refreshed with delay, while alternative could be refreshed immediately, in test you can verify that navigation route is updated in route progress only after its refresh, while refresh of alternative route don't affect route progress nor update navigation route in the progress.

However I don't think it's a blocker.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/7418